docs/project/development-guide.md

```diff
@@ -1,72 +1,436 @@
+  - [Forking the repo](#forking-the-repo)
+  - [Pre-commit Hooks](#pre-commit-hooks)
+  - [Signing off commits](#signing-off-commits)
+  - [Incorporating upstream changes from master](#incorporating-upstream-changes-from-master)
+- [Feast Python SDK / CLI](#feast-python-sdk--cli)
+  - [Environment Setup](#environment-setup)
+  - [Code Style & Linting](#code-style--linting)
+  - [Unit Tests](#unit-tests)
+  - [Integration Tests](#integration-tests)
+    - [Local integration tests](#local-integration-tests)
+    - [(Advanced) Full integration tests](#advanced-full-integration-tests)
+    - [(Advanced) Running specific provider tests or running your test against specific online or offline stores](#advanced-running-specific-provider-tests-or-running-your-test-against-specific-online-or-offline-stores)
+    - [(Experimental) Run full integration tests against containerized services](#experimental-run-full-integration-tests-against-containerized-services)
+  - [Contrib integration tests](#contrib-integration-tests)
+    - [(Contrib) Running tests for Spark offline store](#contrib-running-tests-for-spark-offline-store)
+    - [(Contrib) Running tests for Trino offline store](#contrib-running-tests-for-trino-offline-store)
+    - [(Contrib) Running tests for Postgres offline store](#contrib-running-tests-for-postgres-offline-store)
+    - [(Contrib) Running tests for Postgres online store](#contrib-running-tests-for-postgres-online-store)
+    - [(Contrib) Running tests for HBase online store](#contrib-running-tests-for-hbase-online-store)
+- [(Experimental) Feast UI](#experimental-feast-ui)
+- [Feast Java Serving](#feast-java-serving)
+- [Developing the Feast Helm charts](#developing-the-feast-helm-charts)
+  - [Feast Java Feature Server Helm Chart](#feast-java-feature-server-helm-chart)
+  - [Feast Python / Go Feature Server Helm Chart](#feast-python--go-feature-server-helm-chart)
+- [Feast Go Client](#feast-go-client)
+  - [Environment Setup](#go-environment-setup)
+  - [Building](#building-go)
+  - [Code Style & Linting](#go-code-style--linting)
+  - [Unit Tests](#go-unit-tests)
+  - [Testing with GitHub Actions workflows](#testing-with-github-actions-workflows)
+- [Data Storage Formats](#feast-data-storage-format)
+## Overview
+This guide is targeted at developers looking to contribute to Feast components in
+the main Feast repository:
+- [Feast Python SDK / CLI](#feast-python-sdk--cli)
+- [Feast Java Serving](#feast-java-serving)
+- [Feast Go Client](#feast-go-client)
 
-* [Project Structure](development-guide.md#repository-structure)
-* [Making a Pull Request](development-guide.md#making-a-pull-request)
-* [Feast Data Storage Format](development-guide.md#feast-data-storage-format)
-* [Feast Protobuf API](development-guide.md#feast-protobuf-api)
-* [Maintainer Guide](./maintainers.md)
+Please see [this page](https://docs.feast.dev/reference/codebase-structure) for more details on the structure of the entire codebase.
```

Comment:
nit: use a relative path for consistency

Reply:
done